jsdom-nodejs-env/src/main/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnv.scala

```diff
@@ -49,14 +49,8 @@ class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends JSEnv {
     }
   }
 
-  private def validateInput(input: Input): Unit = {
-    input match {
-      case _:Input.ScriptsToLoad =>
-        // ok
-      case _ =>
-        throw new UnsupportedInputException(input)
-    }
-  }
+  private def validateInput(input: Input): Unit =
+    scriptFiles(input) // for the thrown exception only
```

Comment:
Well, that's not really what I meant. Can't we already calculate whatever we need from `input` here?

Reply:
Yes and no. We can of course extract whatever we need and put it in another data structure that mirrors the supported subset of `Input`. I followed the approach we took in the core `NodeJSEnv`, where we re-match on the `input` at https://github.com/scala-js/scala-js/blob/ff7c2a5b625aca42b96ac3f07c86f57362cf2505/nodejs-env/src/main/scala/org/scalajs/jsenv/nodejs/NodeJSEnv.scala#L146. (although I now see that in that case we simply let the default `MatchError` be thrown, once the input as been validated.

I'm happy to turn it into something that extracts a `ValidatedInput` instead. I'll push another commit.